drivers/at86rf2xx/at86rf2xx.c

```diff
@@ -48,6 +49,8 @@ void at86rf2xx_setup(at86rf2xx_t *dev, const at86rf2xx_params_t *params)
     /* radio state is P_ON when first powered-on */
     dev->state = AT86RF2XX_STATE_P_ON;
     dev->pending_tx = 0;
+    mutex_init(&dev->mutex_sleep);
+    mutex_lock(&dev->mutex_sleep);
```

Comment:
Why do you lock the mutex here?
App never goes after this line
Did you run it on samr21-xpro?

Reply:
The mutex is supposed to block in https://github.com/RIOT-OS/RIOT/pull/8230/files#diff-cf79fb5e4507693e17a065956135c460R129, so it needs to be locked already.

Reply:
So we shouldn't lock it here otherwise it blocks?

Reply:
No. A `mutex_lock()` only blocks if it is called a second time before a `mutex_unlock()` is called (otherwise a mutex would always block ;-)). The same mechanism is used in the current `sema` implementation.

Reply:
@biboc @miri64 is correct, the intention is that the mutex should be always locked and that the second mutex_lock call in the wakeup code will block until the mutex is unlocked from within the interrupt service routine (directly from within ISR context, not from the driver thread since that is blocked by the mutex)
Did you have real problems that the code got stuck on this line or did you not test? How could that happen when the mutex is initialized immediately above (which unconditionally sets the mutex to unlocked state)?

Reply:
@miri64 Yes this is default behaviour of a mutex :-)

@gebart, yes I tried on real hardware and code gets blocked on mutex_lock() which seems very weird because it is initialized just before
Did you try on samr21-xpro hardware? Do you have same problem?

Reply:
I tested very briefly on Mulle. I will test on samr21-xpro later when I have some time.

Reply:
Ok